src/components/shapes/draw.js

```diff
@@ -579,6 +599,208 @@ function setupDragElement(gd, shapePath, shapeOptions, index, shapeLayer, editHe
     }
 }
 
+function drawLabel(gd, index, options, shapeGroup) {
+    // Remove existing label
+    shapeGroup.selectAll('.shape-label').remove();
+
+    // If no label, return
+    if(!options.label) return;
+
+    var labelGroupAttrs = {
+        'data-index': index,
+    };
+    var text = options.label.text;
+    var font = options.label.font;
+
+    var labelTextAttrs = {
+        'data-notex': 1
+    };
+
+    var labelGroup = shapeGroup.append('g')
+        .attr(labelGroupAttrs)
+        .classed('shape-label', true);
+    var labelText = labelGroup.append('text')
+        .attr(labelTextAttrs)
+        .classed('shape-label-text', true)
+        .text(text);
+
+    // If x0, x1, y0, y1 are defined explicitly, use those values
+    // Otherwise, use shape bounding box
+    var shapex0, shapex1, shapey0, shapey1;
+    if('x0' in options && 'x1' in options && 'y0' in options && 'y1' in options) {
+        // Setup conversion functions
+        var xa = Axes.getFromId(gd, options.xref);
+        var xRefType = Axes.getRefType(options.xref);
+        var ya = Axes.getFromId(gd, options.yref);
+        var yRefType = Axes.getRefType(options.yref);
+        var x2p = helpers.getDataToPixel(gd, xa, false, xRefType);
+        var y2p = helpers.getDataToPixel(gd, ya, true, yRefType);
+
+        shapex0 = x2p(options.x0);
+        shapex1 = x2p(options.x1);
+        shapey0 = y2p(options.y0);
+        shapey1 = y2p(options.y1);
+    } else {
+        // Get shape bounding box
+        var shapeBBox = shapeGroup.selectAll('path').node().getBoundingClientRect();
+        shapex0 = shapeBBox.x;
```

Comment:
```suggestion
        shapex0 = shapeBBox.left;
```
So that it does not break IE11.

Reply:
At this point I'm looking for excuses to break IE11 compatibility, but let's say we should make this change for clarity (`x` could mean the left, the right, the middle...) 😅